src/code_8048480.c

```diff
@@ -142,6 +169,136 @@ void sub_80484E8(struct DungeonEntity *r0, struct DungeonEntity *r1)
     {
         SetMessageArgument(gAvailablePokemonNames, r1, 0);
         // Pointer to "But nothing happened!"
-        sub_80522F4(r0, r1, gUnknown_80FB580);
+        sub_80522F4(r0, r1, *gUnknown_80FB580);
+    }
+}
+
+void sub_8048524(struct DungeonEntity *param_1, struct DungeonEntity * param_2)
+{
+  struct DungeonEntityData *iVar2;
+  struct DungeonEntityData *iVar3;
+  u32 *belly;
+  u32 local_14;
+
+  iVar2 = param_2->entityData;
+  iVar3 = iVar2;
+  if (iVar2->isLeader)
+    sub_8078A58(param_1, param_2, 0, 5);
+  else
+  {
+    SetMessageArgument(gAvailablePokemonNames, param_2, 0);
+    if (HasIQSkill(param_2, IQ_SKILL_SELF_CURER))
+        sub_80522F4(param_1, param_2, *gPtrSelfHealPreventedHungerMessage);
+    else
+    {
+      belly = &iVar3->belly;
+      if (RoundUpFixedPoint(*belly) != 0) {
+        sub_80943A0(&local_14, 0);
+        *belly = local_14;
+        sub_80522F4(param_1, param_2, *gUnknown_80F9740);
+      }
+      else
+        sub_80522F4(param_1, param_2, *gUnknown_80F9760);
+    }
+  }
+}
+
+void sub_80485B0(struct DungeonEntity *param_1, struct DungeonEntity * param_2)
+{
+  bool8 isMoveBoosted;
+  s32 moveIndex;
+  struct DungeonEntityData *iVar5;
+  register struct PokemonMove *movePtr1 asm("r4"); // r4
+  register struct PokemonMove *movePtr2 asm("r5"); // r5
+  u8 cVar8;
+  s32 movePowerBoost;
+  s32 moveUnk12;
+
+  isMoveBoosted = FALSE;
+  cVar8 = 1;
+  iVar5 = param_2->entityData;
+  if (DungeonRandomCapped(100) < gUnknown_80F4F46)
+    cVar8 = 3;
+  if (iVar5->isLeader) {
+    for(moveIndex = 0, movePtr1 = &iVar5->moves[0], movePtr2 = movePtr1; moveIndex < MAX_MON_MOVES; movePtr1++, movePtr2++, moveIndex++)
+    {
+        if((movePtr1->moveFlags & MOVE_FLAG_EXISTS) && (movePtr1->moveFlags & MOVE_FLAG_SET))
+        {
+            if(GetMovePower(movePtr2) == 0) continue;
+            movePowerBoost = movePtr1->powerBoost;
+            moveUnk12 = GetMoveUnk12(movePtr2);
```

Comment:
Based on this code, `unk12` is a `maxPowerBoost` for the move, so this can be labeled within the move struct and data. It's 1 for the regular attack and 99 for all other moves.

Reply:
named to `maxPowerBoost` and renamed `GetMoveUnk12` to `GetMoveMaxPowerBoost`